src/smolvm/host/doctor.py

```
@@ -153,6 +153,108 @@ def _check_qemu_version(qemu_path: Path) -> DoctorCheck:
+    rustup_bin = which("rustup") or (
+        str(cargo_home / "bin" / "rustup") if (cargo_home / "bin" / "rustup").exists() else None
+    )
+    if rustup_bin is None and which("cargo") is None:
+        return DoctorCheck(
+            name=name,
+            status="fail",
+            detail="cargo / rustup not found",
+            fix="Install Rust via https://rustup.rs",
+        )
+    try:
+        result = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)